.gitignore

```diff
@@ -15,3 +15,6 @@ mingw_build/
 .vscode
 cmake-build-debug/
 cmake-build-release/
+.cache
+compile_commands.json
+/resources
```

Comment:
The resources directory in the root should not be ignored either; it contains our own generated resources and their project files (and will probably receive more files as time goes on).

I'm not sure where .cache and compile_commands.json come from.

Reply:
.cache and compile_commands.json are for clangd
clangd expects compile_commands.json to be in a parent directory of your sourcefile (I create a symlink to the generated on in `build`).

will put the gitignore changes in a seperate PR